### ISSUE TITLE
Converting some datetime values (without fraction of a second part) produces null values

### ISSUE THREAD

**Issue by vtsyganok-microsoft** [Azure/azure-kusto-python#514]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
            for table in response.iter_primary_results():
                logger.debug(f'Going over first streamed result...')
                df = khelpers.dataframe_from_result_table(table)
```
#### Problem description

For Kusto datetime columns, most of the fields have fraction of a second component, e.g., "2023-12-12T01:59:59.352", which forces panda DataFrame to detect the following format: "%Y-%m-%dT%H:%M:%S.%f%z".

However some of datetime values (approximately 1 out of 1000) fall on .000 mark and KustoStreamingResponseDataSet in this case would have values like "2023-12-12T01:54:44Z" (e.g. omitting fraction of a second part).

If I go to source code ( https://github.com/Azure/azure-kusto-python/blob/cd0ee8a3856e585e98199dbd94a477bbc2438197/azure-kusto-data/azure/kusto/data/helpers.py#L63 ) and change error="coerce" to error="raise" I would be getting the following errors:

```
time data "2023-12-12T01:54:44Z" doesn't match format "%Y-%m-%dT%H:%M:%S.%f%z", at position 1228. You might want to try:
    - passing `format` if your strings have a consistent format;
    - passing `format='ISO8601'` if your strings are all ISO8601 but not necessarily in exactly the same format;
    - passing `format='mixed'`, and the format will be inferred for each element individually. You might want to use `dayfirst` alongside this.
```

Note: with error="coerce" it just produces null value in this case.

There are few possible ways to fix this issue:
- Check if there is no fractional part, forcibly append it
- on this line https://github.com/Azure/azure-kusto-python/blob/cd0ee8a3856e585e98199dbd94a477bbc2438197/azure-kusto-data/azure/kusto/data/helpers.py#L63 add `, format="mixed"` which will force panda to identify datetime format for each row individually.
- on server side always return fraction of a second, even if it's .000

#### Output of ``pip freeze`` 

<details>

[paste the output of ``pip freeze`` here below this line]
adal==1.2.7
applicationinsights==0.11.10
argcomplete==3.2.0
attrs==23.1.0
azure-ai-ml==1.12.1
azure-common==1.1.28
azure-core==1.29.5
azure-graphrbac==0.61.1
azure-identity==1.15.0
azure-keyvault-secrets==4.7.0
azure-kusto-data==4.2.0
azure-kusto-ingest==4.2.0
azure-mgmt-authorization==4.0.0
azure-mgmt-containerregistry==10.3.0
azure-mgmt-core==1.4.0
azure-mgmt-keyvault==10.3.0
azure-mgmt-network==25.1.0
azure-mgmt-resource==23.0.1
azure-mgmt-storage==21.1.0
azure-storage-blob==12.19.0
azure-storage-file-datalake==12.14.0
azure-storage-file-share==12.15.0
azure-storage-queue==12.9.0
azureml-core==1.54.0.post1
azureml-dataprep==4.12.7
azureml-dataprep-native==38.0.0
azureml-dataprep-rslex==2.19.6
azureml-dataset-runtime==1.54.0.post1
azureml-mlflow==1.54.0.post1
azureml-telemetry==1.54.0
backports.tempfile==1.0
backports.weakref==1.0.post1
bcrypt==4.1.1
cachetools==5.3.2
certifi==2023.11.17
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
cloudpickle==2.2.1
colorama==0.4.6
contextlib2==21.6.0
contourpy==1.2.0
cryptography==41.0.7
cycler==0.12.1
databricks-cli==0.18.0
distro==1.8.0
docker==6.1.3
dotnetcore2==3.1.23
entrypoints==0.4
fonttools==4.46.0
gitdb==4.0.11
GitPython==3.1.40
google-api-core==2.15.0
google-auth==2.25.2
googleapis-common-protos==1.62.0
humanfriendly==10.0
idna==3.6
ijson==3.2.3
importlib-metadata==7.0.0
isodate==0.6.1
jeepney==0.8.0
jmespath==1.0.1
joblib==1.3.2
jsonpickle==3.0.2
jsonschema==4.20.0
jsonschema-specifications==2023.11.2
kiwisolver==1.4.5
knack==0.11.0
llvmlite==0.41.1
marshmallow==3.20.1
matplotlib==3.8.2
missingno==0.5.2
mlflow-skinny==2.9.1
msal==1.26.0
msal-extensions==1.0.0
msrest==0.7.1
msrestazure==0.6.4
ndg-httpsclient==0.5.1
numba==0.58.1
numpy==1.23.5
oauthlib==3.2.2
opencensus==0.11.3
opencensus-context==0.1.3
opencensus-ext-azure==1.1.12
packaging==23.2
pandas==2.1.4
paramiko==3.3.1
pathspec==0.12.0
Pillow==10.1.0
pkginfo==1.9.6
portalocker==2.8.2
protobuf==4.25.1
psutil==5.9.6
pyarrow==12.0.1
pyasn1==0.5.1
pyasn1-modules==0.3.0
pycparser==2.21
pydash==7.0.5
Pygments==2.17.2
PyJWT==2.8.0
PyNaCl==1.5.0
pyOpenSSL==23.3.0
pyparsing==3.1.1
pyreadline3==3.4.1
PySocks==1.7.1
python-dateutil==2.8.2
pytz==2023.3.post1
pywin32==306
PyYAML==6.0.1
referencing==0.32.0
requests==2.31.0
requests-oauthlib==1.3.1
rpds-py==0.13.2
rsa==4.9
scikit-learn==1.3.2
scipy==1.11.4
seaborn==0.13.0
SecretStorage==3.3.3
shap==0.44.0
six==1.16.0
slicer==0.0.7
smmap==5.0.1
sqlparse==0.4.4
strictyaml==1.7.3
tabulate==0.9.0
tenacity==8.2.3
threadpoolctl==3.2.0
tqdm==4.66.1
typing_extensions==4.9.0
tzdata==2023.3
urllib3==2.1.0
websocket-client==1.7.0
xgboost==2.0.2
zipp==3.17.0
</details>

**Comment by AsafMah** [comment]
Fixed 4.3.1
